### ISSUE TITLE
fresh install on debian - redirector problems

### ISSUE THREAD

**Issue by orkim** [hlandau/acme#128]
I did a fresh install on a debian box via the apt repo.  I ran through the quickstart, enabled the redirector (my http server only listens on 443), and issued a 'want' for my domain.  Everything worked great and even port 80->443 redirection was working.

I manually issued a "service acmetool-redirector stop" and did some tinkering to the http server.  Now trying to restart the redirector service via "acmetool redirector" or the service "service acmetool-redirector start" gives me the following error.

20160222161931 [CRITICAL] acmetool: fatal: storage: unknown certificate: "csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa"

I do have a /var/lib/acme/certs/csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa directory with cert/chain/fullchain/privkey/url within it.

This seems to come from [this line](https://github.com/hlandau/acme/blob/bc6a080336e0767b4bb02d188ddc5b671143f644/storage/storage-fdb.go#L137) which says "this should never happen" in the comment above.

Could you help me understand where this error is coming from?

**Comment by hlandau** [comment]
Can you post the output of `acmetool --xlog.severity=trace`?

**Comment by orkim** [comment]
Here ya go.

```
20160222163456 [DEBUG] fdb: enforce permissions: tmp/symlink.749006260 0/0 0/0
20160222163456 [DEBUG] acme.storageops: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) satisfies Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [TRACE] acme.storageops: findBestCertificateSatisfying: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) > <nil>
20160222163456 [DEBUG] acme.storageops: Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0): best certificate satisfying is Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa), err=<nil>
20160222163456 [DEBUG] acme.storageops: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) needsRenewing=false notAfter=2016-05-22 21:08:00 +0000 UTC
20160222163456 [DEBUG] acme.storageops: Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0): have best certificate which does not need renewing, skipping target
20160222163456 [DEBUG] acme.storageops: done processing targets, reconciliation complete, 0 errors occurred
20160222163456 [DEBUG] fdb: enforce permissions: tmp/symlink.055649667 0/0 0/0
20160222163456 [DEBUG] acme.storageops: disjoint hostname mapping: mail.6khz.com -> Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [DEBUG] acme.storageops: disjoint hostname mapping: 6khz.com -> Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [DEBUG] acme.storageops: disjoint hostname mapping: www.6khz.com -> Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [DEBUG] acme.storageops: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) satisfies Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [TRACE] acme.storageops: findBestCertificateSatisfying: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) > <nil>
20160222163456 [TRACE] acme.storageops: relink: best certificate satisfying Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0) is Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa)
20160222163456 [DEBUG] acme.storageops: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) satisfies Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [TRACE] acme.storageops: findBestCertificateSatisfying: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) > <nil>
20160222163456 [TRACE] acme.storageops: relink: best certificate satisfying Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0) is Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa)
20160222163456 [DEBUG] acme.storageops: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) satisfies Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0)
20160222163456 [TRACE] acme.storageops: findBestCertificateSatisfying: Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa) > <nil>
20160222163456 [TRACE] acme.storageops: relink: best certificate satisfying Target(6khz.com,www.6khz.com,mail.6khz.com;https://acme-v01.api.letsencrypt.org/directory;0) is Certificate(csdj2fxa6ag4y5yqwcpgojoarnzf32xkj2r7x3ctpeo3z6gpygqa)
```

**Comment by hlandau** [comment]
Bug confirmed. Fix imminent.

**Comment by hlandau** [comment]
Fixed in v0.0.49. PPA should be updated within 30 minutes. Thanks for the report.
